### ISSUE TITLE
Node v13 causes builds to fail

### ISSUE THREAD

**Issue by ddinchev** [edbizarro/gitlab-ci-pipeline-php#74]
For the past hour our builds on GitLab are failing due to incompatibility of node-sass with Node v13. 

Given that Node v13 is not an LTS version, do we really need to jump to it as soon as it comes out? Or use odd versions at all? There [were similar problems with v12](https://github.com/edbizarro/gitlab-ci-pipeline-php/issues/60), as `gitlab-ci-pipeline-php` was updated to it as soon as it came out. Isn't it reasonable even for LTS versions to wait for a month or so as glitches are dealt with or maybe maintain a stable version of this repo?

Don't get me wrong - I think this is a terrific project and makes it super easy to have PHP builds in GitLab CI. I'm skeptical about the value the bleeding edge brings while the problems that come with it are very real.

**Comment by edbizarro** [comment]
hi @ddinchev 

You're right, actually, I don't have a pattern or schedule to update the images, from time to time I check the https://nodejs.org/en/ and if there's a new version I update the images, but I think your suggestion to wait some time before upgrade the node is very good, I'll pay more attention to the release dates and be more cautious in the future.

**Comment by ddinchev** [comment]
Thanks for your quick reply! I'd suggest we stick to LTS versions - which are even numbers, eg v8, v10, v12 and in 6 months v14. Odd major versions like v9, v11, v13 are more likely to contain new features and breaking changes - until the LTS version comes, glitches are fixed but it still doesn't hurt to wait a month after the new release before switching over.

Meanwhile - there is a `node-sass:4.13` version, supporting v13.

**Comment by edbizarro** [comment]
I've pushed some changes with node v12 again and I'll wait a few weeks/months to update again to v13.

Thanks for your suggestion and sorry for the trouble

**Comment by pfaffenrodt** [comment]
@edbizarro 
we also face some build fails caused by sass and node-gyp trying to compile c++ files.

It may be nice, if we could use "nvm". Maybe with preinstalled versions of nodejs with nvm, so builds are faster. In my usecase I just fixed an typo an project and my build failed due the update of nodjs. So if some project is stuck to an old version and you dont realy have time to face this issue. I would prefer to fix a pipeline to a specifc nodejs version for those builds. Projects define and use the nodejs version that has worked and should work in future.

basically my idea is to support following:
```yaml
scripts: 
- nvm install 10.16.3
- nvm use 10.16.3
```

**Comment by aleksblendwerk** [comment]
nvm and possibly preinstalled previous node versions would be a great addition. Even though I see not everyone will need this and it will enlarge the image.

I am currently in need of an image with PHP 7.2 and node 10.x. For now I solved it by installing nvm, node 10 and yarn manually on top of the 7.2 image... or is there a better yet still convenient way?

**Comment by pfaffenrodt** [comment]
I thought about spliting my pipeline and do an extra job using an node image and pass all assets to the next job as artifact.
